### PR TITLE
Split the encoder and estimation velocity of motor

### PR DESCRIPTION
**Pull request by IhorNehrutsa**
I have separated the encoder and the estimation velocity of motor.
Estimation velocity is the best choice.
0RPM
P10714-140653
375RPM
P10714-140755

### CODE REVIEW

File: src/hardware/encoder.cpp [hunk 1]
```diff
@@ -577,21 +576,20 @@ double Encoder::getSpeed() {
...
     return (micros() - lastAngleSampleTime > SPD_EST_MIN_INTERVAL);
 }
-
-
-#else // !ENCODER_SPEED_ESTIMATION
+#endif
 
 int16_t Encoder::getRawSpeed() {
-
+#if 0
```
Comment (CAP1Sup) [inline src/hardware/encoder.cpp:592]: Why not just remove this? Then it'll be cleaner

File: src/hardware/encoder.cpp [hunk 2]
```diff
@@ -620,12 +634,21 @@ double Encoder::getSpeed() {
 
 	// Get raw speed reading
 	int16_t rawSpeed = rawData[0];
-	rawSpeed = rawSpeed & DELETE_BIT_15;
+    #if 0
```
Comment (CAP1Sup) [inline src/hardware/encoder.cpp:637]: Same as above

File: src/hardware/encoder.cpp [hunk 3]
```diff
@@ -686,7 +707,7 @@ double Encoder::getAccel() {
 
 // Reads the raw momentary temperature of the encoder
 int16_t Encoder::getRawTemp() {
-
+#if 0
```
Comment (CAP1Sup) [inline src/hardware/encoder.cpp:710]: Same as above pt2

File: src/hardware/encoder.cpp [hunk 4]
```diff
@@ -761,7 +799,7 @@ double Encoder::getTemp() {
 
 // Gets the raw revolutions from the motor in range [-258 ... +257]
 int16_t Encoder::getRawRev() {
-
+#if 0
```
Comment (CAP1Sup) [inline src/hardware/encoder.cpp:802]: Needs checked

**Comment by IhorNehrutsa** [comment]
```
#if 0
```
It's more convenient for now, then you can delete it.
It is also a comment, >> and << are not obvious.

File: src/software/macros.h [hunk 5]
```diff
@@ -37,7 +37,7 @@
 // Direction (for reverse direction)
 #define DIRECTION(x) ((x) > 0 ? 1 : (-1))  // Note: zero 'x' is not allowed
 
-// Return angle in range [0..360]
+// Return angle in range [0..360[
```
Comment (CAP1Sup) [inline src/software/macros.h:40]: Sorry, I thought that this was a mistype

**Pull request merged**